tests/source/test_file.py

```diff
@@ -76,6 +84,13 @@ def test_repr(self):
             await source.close()
             m_open.assert_called_once_with('testfile', 'w')
 
+    async def test_close_gz(self):
+        source = FakeFileSource('testfile.gz')
+        m_open = mock_open()
+        with patch('builtins.open', m_open):
+            await source.close()
+            m_open.assert_called_once_with('testfile.gz', 'wb')
```

Comment:
This is a much better idea!!

Reply:
Is there anything else to be done for gzip support? Or shall I start working on the next?